pymatgen/analysis/graphs.py

```diff
@@ -1489,6 +1490,8 @@ def get_subgraphs_as_molecules(self, use_weights: bool = False) -> list[Molecule
         # without adding extra logic
         if getattr(self, "_supercell_sg", None) is None:
             self._supercell_sg = supercell_sg = self * (3, 3, 3)
+        else:  # TODO: need double check
+            raise RuntimeError("Supercell spacegroup is not None.")
```

Comment:
This condition is weird, need some expert advice on this, this function only works when `self._supercell_sg` is `None`:

https://github.com/materialsproject/pymatgen/blob/ad6eafee2474718813cba57aa8e1752f1181b639/pymatgen/analysis/graphs.py#L1467-L1544

Reply:
I just looks into the history, it was added by @mkhorton in #831 (line 937), and the original code is written this way. Maybe we should keep it as is for now?